daemon/k8s_watcher.go

```diff
@@ -1429,6 +1432,36 @@ func (d *Daemon) deleteCiliumNetworkPolicyV2(cnp *cilium_v2.CiliumNetworkPolicy)
 	}
 }
 
+func updateRuleAnnotations(newRuleCpy *cilium_v2.CiliumNetworkPolicy) error {
+
+	// Create DeepCopy, as newRuleCpy is a pointer and may be modified elsewhere
+	// outside of this function, and we want to ensure that the annotations
+	// which are extracted are not modified.
+	ruleCopyForAnnotations := newRuleCpy.DeepCopy()
+
+	// Only update annotations for node on which this agent is running.
+	nodeName := node.GetName()
+	cnpns := ruleCopyForAnnotations.GetPolicyStatus(nodeName)
+	cnpns.Annotations = ruleCopyForAnnotations.Annotations
+
+	// Do not update annotations if policy rule failed to be installed previously.
+	if !cnpns.OK {
```

Comment:
What you probably want here is `cnpns.Enforcing` and not `cnpns.OK`

Reply:
Ah, yes, you're right. Fixed!